author	Hao Chen <chenhao288@hisilicon.com>	2021-11-18 20:12:43 +0800
committer	David S. Miller <davem@davemloft.net>	2021-11-22 12:31:49 +0000
commit	7462494408cd3de8b0bc1e79670bf213288501d0 (patch)
tree	61a9e0eb7e7f2e53c2afca17e284dcef3f85d572
parent	0b70c256eba8448b072d25c95ee65e59da8970de (diff)
download	linux-7462494408cd.tar.gz
ethtool: extend ringparam setting/getting API with rx_buf_len

Add two new parameters kernel_ringparam and extack for
.get_ringparam and .set_ringparam to extend more ring params
through netlink.

Signed-off-by: Hao Chen <chenhao288@hisilicon.com>
Signed-off-by: Guangbin Huang <huangguangbin2@huawei.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 src/i40e_ethtool.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/i40e_ethtool.c b/src/i40e_ethtool.c
index 456e59c..f80c7d8 100644
--- a/src/i40e_ethtool.c
+++ b/src/i40e_ethtool.c
@@ -2227,7 +2227,9 @@ static void i40e_get_drvinfo(struct net_device *netdev,
 }
 
 static void i40e_get_ringparam(struct net_device *netdev,
-			       struct ethtool_ringparam *ring)
+			       struct ethtool_ringparam *ring,
+                               struct kernel_ethtool_ringparam *kernel_ring,
+                               struct netlink_ext_ack *extack)
 {
 	struct i40e_netdev_priv *np = netdev_priv(netdev);
 	struct i40e_pf *pf = np->vsi->back;
@@ -2255,7 +2257,9 @@ static bool i40e_active_tx_ring_index(struct i40e_vsi *vsi, u16 index)
 }
 
 static int i40e_set_ringparam(struct net_device *netdev,
-			      struct ethtool_ringparam *ring)
+			      struct ethtool_ringparam *ring,
+                              struct kernel_ethtool_ringparam *kernel_ring,
+                              struct netlink_ext_ack *extack)
 {
 	struct i40e_ring *tx_rings = NULL, *rx_rings = NULL;
 	struct i40e_netdev_priv *np = netdev_priv(netdev);
-- 
1.8.3.1

